src/components/PostDetails.tsx

```diff
@@ -1,117 +1,127 @@
-import React from 'react';
+import React, {
+  Dispatch,
+  SetStateAction,
+  useState,
+  useEffect,
+} from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { deleteComment } from '../api/comments';
+import { warningTimer } from '../utils/warningTimer';
+
+type Props = {
+  post: Post | null;
+  comments: Comment[];
+  setComments: Dispatch<SetStateAction<Comment[]>>;
+  isLoaderComments: boolean,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  post,
+  comments,
+  setComments,
+  isLoaderComments,
+}) => {
+  const [isWriteComment, setIsWriteComment] = useState(false);
+  const [isHasError, setIsHasError] = useState(false);
+
+  const removeComment = async (commentRemove: Comment) => {
```

Comment:
pass only data you actually need
```suggestion
  const removeComment = async (id: string) => {
```